src/event.rs

```diff
@@ -236,3 +240,28 @@ impl<N: Notify> Processor<N> {
         self.mouse_bindings = config.mouse_bindings().to_vec();
     }
 }
+
+trait EventTranslator {
+    fn translate_event(self, glutin::Event) -> glutin::Event;
+}
+
+struct ModTranslator {
+    mods_to_match: glutin::Mods,
+    mods_to_add: glutin::Mods,
+}
+
+impl EventTranslator for ModTranslator {
+    fn translate_event(self, event: glutin::Event) -> glutin::Event {
```

Comment:
I do appreciate the abstraction here, but matching twice on `event` seems less than ideal. Perhaps we can just apply this to mods directly.